example-raft-kv/src/network/api.rs

```diff
@@ -20,9 +22,57 @@ use crate::store::ExampleRequest;
  */
 #[post("/write")]
 pub async fn write(app: Data<ExampleApp>, req: Json<ExampleRequest>) -> actix_web::Result<impl Responder> {
+    // Redirect this call to the leader, this is required because learners are not allowed to write data to
+    // other nodes. 
+    if let Some(result) = redirect_to_leader_if_required(&app, &req, "/write").await {
+        return result;
+    }
     let request = ClientWriteRequest::new(EntryPayload::Normal(req.0));
     let response = app.raft.client_write(request).await;
-    Ok(Json(response))
+    Ok(if response.is_ok() { 
+        HttpResponse::Ok().finish()
+    } else { 
+        HttpResponse::BadRequest().body("Failed write number")
+    })
+}
+
+async fn redirect_to_leader_if_required<T: Serialize>(
+  app: &Data<ExampleApp>,
+  req: &Json<T>,
+  route: &str,
+) -> Option<actix_web::Result<HttpResponse>> {
+
+  let metrics = app.raft.metrics();
+  let metrics = metrics.borrow();
+
+  if let Some(current_leader) = metrics.current_leader {
+
+      if metrics.id == current_leader {
+          return None;
+      }
```

Comment:
Checking against metrics does not guarantee 100% accuracy:
The leader may have already changed to another role.

The correct way is to write to raft and to see if it returns `ForwardToLeader` Error.